core/lib/zksync_core/src/eth_watch/event_processors/governance_upgrades.rs

```diff
@@ -0,0 +1,117 @@
+use crate::eth_watch::{
+    client::{Error, EthClient},
+    event_processors::EventProcessor,
+};
+use std::convert::TryFrom;
+use std::time::Instant;
+use zksync_contracts::governance_contract;
+use zksync_dal::StorageProcessor;
+use zksync_types::{
+    protocol_version::GovernanceOperation, web3::types::Log, Address, ProtocolUpgrade,
+    ProtocolVersionId, H256,
+};
+
+/// Responsible for saving new protocol upgrade proposals to the database.
```

Comment:
I'd explain more - that it listens on the events coming from a given operations from the governance contract - and then saves them in a database.